src/SubjectPreview.js

```diff
@@ -1,44 +1,164 @@
-                            color="textPrimary"
-                        >
-                            {subject.code}
-                        </Typography>
-                        {" — " + subject.lecturer}
-                        <br />
-                        <Typography
-                            component="span"
-                            variant="body2"
-                        >
-                            {"Rozvrhováno: " + formatDay(subject.day)}
-                        </Typography>
-                    </React.Fragment>
-                }/>
+        <ListItem key={subject.code} className={classes.root}>
+            <ListItemText primary={<SubjectDialog subject={subject}/>}/>
+
+            <Typography component="div" variant="body2" color="textSecondary">
+            <Grid container spacing={0} justify="space-between">
+                <Grid container item xs={12} sm={9} spacing={1} justify="space-between">
+                    <Grid item xs={10} sm={6}>
+                        <ItemFirstLine subject={subject}/>
+                    </Grid>
+                    <Grid item xs={6} sm={6}>
+                        <ItemTimetable subject={subject} align="right"/>
+                    </Grid>
+
+                    <Grid item xs={9} sm={12}>
+                        <ItemAnnotation subject={subject}/>
+                    </Grid>
+                </Grid>
+                <Grid container item xs={12} sm={3} spacing={1}>
+                    <Grid item xs={12}>
+                        <ItemExamType subject={subject}/>
+                    </Grid>
+                    <Grid item xs={12}>
+                        <ItemCredits subject={subject} useBar/>
+                    </Grid>
+                    <Grid item xs={12}>
+                        <ItemHours subject={subject} useBar/>
+                    </Grid>
+                </Grid>
+            </Grid>
+            </Typography>
         </ListItem>
     );
 };
 
-export default SubjectPreview;
\ No newline at end of file
+SubjectPreview.propTypes = {
+    subject: PropTypes.shape({
+        code: PropTypes.string.isRequired,
+        lecturer: PropTypes.string.isRequired,
+        // day: PropTypes.number,  // TODO Should be number and not english string as it is now (toy data)
+        time: PropTypes.string.isRequired,
+        credits: PropTypes.number.isRequired,
+        len: PropTypes.string.isRequired,  // TODO Number or string? ... "2+2"?
+
+        // TODO Following things are not yet in toy data
+        // name: PropTypes.string.isRequired,
+        // department: PropTypes.string.isRequired,
+        // departmentCode: PropTypes.string.isRequired,
+        // examType: PropTypes.string.isRequired,
+        // annotation: PropTypes.string.isRequired,
+    })
+};
+
+export default SubjectPreview;
+
+
+// TODO Probably should move these definitions somewhere else in the future
+const ItemFirstLine = ({subject}) => {
+    return (
```

Comment:
Tady se opakuje stejný pattern. Napíšu k tomu něco zde. Řádky
```
const ItemFirstLine = ({subject}) => {
    return (
```
jsou vlastně úplně zbytečné, protože tělo toho componentu není o moc delší. Na druhou stranu chápu, že jsi to pěkně separoval a je to přehledné. Myslím, že vy výsledku je to otázka preference.

Osobně mi ty componentíky od řádku 92 níž připadají zbytečné, akorát si musím pamatovat další jména a skipovat nahoru dolu, když čtu kód.

Neber to jako, že jsem proti.  Mít kód, který je modulární, ale zase ne přespříliš rozsekaný je těžké a často je to otázka preference.

Reply:
Dávat do jiného souboru bych to ve výsledku taky nechtěl, udělal jsem to takto a akorát jsem nevěděl, jestli to neútočilo na nějaké zásadní pattern pravidlo. Mě se s tím takhle, s abstrakcí pro každý ten kousek, líp pracovalo. Když člověk chce změnit vykreslování kreditů tak zascrolluje dolů, když řeší pořadí jednotlivých kousků, tak zůstane nahoře. Když se to naplácá všechno do sebe tak je to trošku taková hrouda, je to přece jen už větší komponent.

Zatím to v rámci cesty nejmenšího odporu nechám takhle :)

Reply:
Jasný a přesně o tom to je, tady už je to opravdu down to preference.